### PR TITLE
refactor(cli): `once_cell::sync::Lazy` -> `std::sync::LazyLock`, `once_cell::sync::OnceCell` -> `std::sync::OnceLock`

### PR DESCRIPTION
**Pull request by linbingquan**
refactor(cli): `once_cell::sync::Lazy` -> `std::sync::LazyLock`, `once_cell::sync::OnceCell` -> `std::sync::OnceLock`

link: https://blog.rust-lang.org/2024/07/25/Rust-1.80.0.html#lazycell-and-lazylock

### CODE REVIEW

File: cli/lsp/tsc.rs [hunk 1]
```diff
@@ -94,19 +93,28 @@ use tower_lsp::jsonrpc::Error as LspError;
 use tower_lsp::jsonrpc::Result as LspResult;
 use tower_lsp::lsp_types as lsp;
 
-static BRACKET_ACCESSOR_RE: Lazy<Regex> =
-  lazy_regex!(r#"^\[['"](.+)[\['"]\]$"#);
```
[the remark this answers is not in the document]
Comment (linbingquan) [reply cli/lsp/tsc.rs:98]: > Why this change? `lazy_regex` evaluates the regex at compile time from my understanding and now we're no longer doing that.

Because it has this error.  Correct me if I am wrong.

```bash
error[E0308]: mismatched types
  --> cli\lsp\tsc.rs:98:20
   |
98 |   LazyLock::new(|| lazy_regex!(r#"^\[['"](.+)[\['"]\]$"#));
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Regex`, found `Lazy<_, ...>`
   |
   = note: expected struct `regex::Regex`
              found struct `lazy_regex::Lazy<_, {closure@cli\lsp\tsc.rs:98:20: 98:58}>`
   = note: this error originates in the macro `lazy_regex` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0308`.
error: could not compile `deno` (bin "deno") due to 1 previous error
```

[the remark this answers is not in the document]
Comment (dsherret) [reply cli/lsp/tsc.rs:98]: I think it can be changed to:

```rs
static BRACKET_ACCESSOR_RE: lazy_regex::Lazy<Regex> =
  lazy_regex!(r#"^\[['"](.+)[\['"]\]$"#);
```

**Comment by bartlomieju** [comment]
@linbingquan if you're still interested in landing this PR can you please rebase it and apply David's suggestion from https://github.com/denoland/deno/pull/25104#discussion_r1725073239?

**Comment by linbingquan** [comment]
> @linbingquan if you're still interested in landing this PR can you please rebase it and apply David's suggestion from [#25104 (comment)](https://github.com/denoland/deno/pull/25104#discussion_r1725073239)?

I saw a lot of code that needed to be cut before 2.0 was released, so I didn't do it.

To be honest, I upgraded the `Rust 1.81.0` build after the 2.0 release and it failed...

Let me try again.

**Comment by linbingquan** [comment]
Note: `cli\factory.rs` and `cli\standalone\mod.rs` also use `once_cell::sync::OnceCell`, I am not change it.

**Review by bartlomieju — COMMENTED** [review]
Would it be possible to remove dependency on `once_cell` in various crates you changed? Ie. remove it from relevant `Cargo.toml` files.

**Comment by linbingquan** [comment]
> Ie. remove it from relevant `Cargo.toml` files.

I have a quick look, should not be able to complete the deletion of 'once_cell', 'once_cell' is used elsewhere.

> Would it be possible to remove dependency on `once_cell` in various crates you changed?

I'm sorry, I didn't know what you wanted me to do.

**Comment by linbingquan** [comment]
I now feel that this refactoring is not very necessary, I close this PR.
